### PR TITLE
fix: harden release versioning and installer fallback behavior

### LINKED ISSUE #555: [BUG] release mismatching between actual release version and installed release version
### Summary

The published Windows release asset for v2026.4.13 contains an opensre.exe binary that reports 2026.4.5 instead of 2026.4.13.

### Expected vs actual behavior

Expected:
The v2026.4.13 Windows release asset (opensre_2026.4.13_windows-x64.zip) should contain an opensre.exe binary that reports version 2026.4.13.

Actual:
The published ZIP contains an opensre.exe that reports:
`
opensre, version 2026.4.5
`

This also breaks the current installer validation, because the installer expects 2026.4.13 but the downloaded binary reports 2026.4.5.

### Steps to reproduce

1. On Windows, download the published asset: https://github.com/Tracer-Cloud/opensre/releases/download/v2026.4.13/opensre_2026.4.13_windows-x64.zip
2. Extract the ZIP.
3. `.\opensre.exe --version`
4. Observe that it prints `opensre, version 2026.4.5`

### Can you reproduce it consistently?

Yes

### How often does it occur?

Every time

### Operating system

Windows

### Logs and error output

I verified this by directly downloading and extracting the published Windows asset from the v2026.4.13 release. This is reproducible without relying on local PATH resolution, so it appears to be a release artifact/versioning problem rather than only a local install-path issue.

### Additional context

check the installing version and actual installed version 
Image

### PR DESCRIPTION
**Pull request by sundaram2021**
Fixes #555

#### Describe the changes you have made in this PR -

- Synced release builds to the Git tag version before building wheels, sdists, and frozen binaries.
- Added release workflow checks to verify artifact filenames and binary `--version` output match the tagged release version.
- Hardened `install.ps1` and `install.sh` so `latest` installs can recover when GitHub release metadata and the checksum-verified binary version drift, while keeping pinned `OPENSRE_VERSION` installs strict.
- Added a Bash guard to `install.sh` and enforced LF line endings for `*.sh`.
- Updated `app/remote/system_metrics.py` to handle Unix-only metrics APIs safely on Windows, which fixes the Windows mypy failures.

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [ ] No, I wrote all the code myself
- [x] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [x] I have reviewed every single line of the AI-generated code
- [x] I can explain the purpose and logic of each function/component I added
- [x] I have tested edge cases and understand how the code handles them
- [x] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**

This PR fixes a release drift problem where the release tag could advance independently of the package/binary version, which caused the installers to fail on version verification. I considered simply relaxing installer validation, but that would have hidden real release problems and weakened pinned installs. Instead, I made the release workflow use the tag as the source of truth before building artifacts, and I kept installers strict for pinned versions while allowing `latest` installs to warn and continue when the downloaded binary is checksum-verified but the GitHub release metadata is inconsistent.

The key pieces are:
- `packaging/sync_release_version.py`, which updates `pyproject.toml` and `app/version.py` from the release tag.
- `.github/workflows/release.yml`, which runs that sync step before building Python distributions and binaries, and verifies the resulting artifacts report the expected version.
- `install.ps1` and `install.sh`, which now recover gracefully from the currently broken `latest` release metadata case without masking real pinned-version mismatches.
- `app/remote/system_metrics.py`, which now guards Unix-only APIs so Windows environments degrade gracefully and type-check cleanly.

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [ ] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [ ] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -246,16 +251,25 @@ verify_binary_version() {
...
+      printf '%s\n' "$expected_version"
       ;;
     *)
-      die "Downloaded binary version mismatch. Expected '${expected_version}' but got: ${version_output}"
+      if [ -n "${OPENSRE_VERSION:-}" ] || [ -z "$actual_version" ]; then
+        die "Downloaded binary version mismatch. Expected '${expected_version}' but got: ${version_output}"
+      fi
+
+      warn "Latest release metadata reports v${expected_version}, but the downloaded binary reports v${actual_version}. Installing the verified binary anyway."
+      printf '%s\n' "$actual_version"
```
[the remark this answers is not in the document]
Comment (sundaram2021) [reply install.sh:272]: we are not using this function anywhere , LGTM

File: packaging/sync_release_version.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+import argparse
+import re
+from collections.abc import Callable
+from pathlib import Path
+
+ROOT = Path(__file__).resolve().parents[1]
+PYPROJECT_PATH = ROOT / "pyproject.toml"
+APP_VERSION_PATH = ROOT / "app" / "version.py"
+VERSION_PATTERN = re.compile(r"^v?(?P<version>\d{4}\.\d{1,2}\.\d{1,2})$")
```
[the remark this answers is not in the document]
Comment (sundaram2021) [reply packaging/sync_release_version.py]: done

File: packaging/sync_release_version.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+def _replace_project_version(version: str, text: str) -> str:
+    lines = text.splitlines(keepends=True)
+    in_project_section = False
+
+    for index, line in enumerate(lines):
+        stripped = line.strip()
+
+        if stripped.startswith("[") and stripped.endswith("]"):
+            in_project_section = stripped == "[project]"
```
[the remark this answers is not in the document]
Comment (sundaram2021) [reply packaging/sync_release_version.py:36]: resolved

**Comment by Devesh36** [comment]
Thankyou @sundaram2021
